pkg/compose/watch.go

```diff
@@ -183,15 +183,14 @@ func (s *composeService) watch(ctx context.Context, syncChannel chan bool, proje
 	}
 	options.LogTo.Log(api.WatchLogger, "Watch enabled")
 
-	err = eg.Wait()
 	for {
 		select {
 		case <-ctx.Done():
-			return err
+			return eg.Wait()
 		case <-syncChannel:
 			options.LogTo.Log(api.WatchLogger, "Watch disabled")
 			ctx.Done()
```

Comment:
ctx.Done() here is weird, this does nothing (return the done channel, this doesn't stop context)